### DISCUSSION TITLE
_Gtk3 _Gtk4 Gtk and _Gdk3 _Gdk4 Gdk?

### DISCUSSION THREAD

**Discussion by acmpo6ou** [pygobject/pygobject-stubs#130] [Q&A] [answered]
Hello! Thank you for these awesome stubs for PyGObject :heart: 

I noticed after v2.0.0 you added `_Gtk3.pyi` and `_Gtk4.pyi`, also `Gtk.pyi` doesn't have all classes that it used to. I can solve this by using `_Gtk3.pyi`.

But I was wondering why do you have `_Gtk3.pyi` and `_Gtk4.pyi`? I can imagine this is because PyGObject is adopting Gtk4? Is it already possible to try out Gtk4 in PyGObject? Maybe you have some resources with the news about Gtk4 and PyGObject.

Thank you! :+1:  :+1:

**Comment by lovetox** [comment, marked as answer]
hi, the README describes the configuration options available.

If you don’t configure anything, the most current version of each lib is installed, this means for example Gtk4, thats probably why you think classes are missing. They simply don’t exist anymore in Gtk4

**Reply by acmpo6ou** [reply]
Thanks a lot! So silly of me not to check the README first 🤦🏻 I'm so sorry for this 😇

**Reply by boardkeystown** [reply]
yeah I was losing my mind on this until I read the readme also.


I'm sure someone will end up in the same situation as me.


You are working with PyGOobject Gtk and realize it does not have auto complete b/c it is a shared object.

You google mypy and stubs and find PyGObject-stubs on pypi and pip install

Then most things don't work.


All you gotta do is pip unisntall pygobject-stubs 

then set the environment var

PYGOBJECT_STUB_CONFIG=Gtk3,Gdk3,Soup2

if you are working with Gtk3

then 
  
pip install pygobject-stubs 

Also the generate py too you have is cool.

Sure someone else will run into the same issue.
